### DISCUSSION TITLE
How to cache queries

### DISCUSSION THREAD

**Discussion by cpinflux** [grafana/grafana-infinity-datasource#799] [Q&A]
I've got some alerts and dashboard variables for which I'd like to cache the query data for a reasonable length of time. The docs (mostly) say that caching is generically supported, but under Grafana Enterprise 10.3.4 I see nothing about caching at any point with this plugin. General config says caching is enabled in [caching] but past that, no sign for this datasource. for the I feel like I'm missing something obvious and fundamental, but ... any clues?

**Comment by yesoreyeram** [comment]
Hi @cpinflux - plugin itself doesn't have any built-in caching. But it supports the [grafana enterprise query caching](https://grafana.com/docs/grafana/latest/administration/data-source-management/#query-and-resource-caching) which is available only to grafana enterprise/cloud customers.

you can configure caching in the datasource config page

<img width="827" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/56fadc98-071d-4d66-a68e-05de991452df">

**Reply by PinguDEV-original** [reply]
any other way?

**Reply by yesoreyeram** [reply]
sorry buddy. not aware of other ways.

**Comment by dal00** [comment]
Depending on what you're doing you could try out this plugin which is kind of EOL but it does have a cache. 

https://github.com/grafana/grafana-json-datasource

https://grafana.com/grafana/plugins/marcusolsson-json-datasource/

**Comment by zachsents** [comment]
Kinda BS caching is an enterprise feature
